### PR TITLE
add forogger task

### PR DESCRIPTION
**Pull request by frostwolm**
### Object-Oriented JavaScript

[Demo](https://frostwolm.github.io/frogger/) |
[Code base](https://github.com/frostwolm/frogger)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/frostwolm/frogger/js/app.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    if (collisionX && collisionY) {
+      player.dead();
+    }
+};
+
+// Now write your own player class
+// This class requires an update(), render() and
+// a handleInput() method.
+var Player = function () {
+  this.x = 2 * 101;
```
Comment (OleksiyRudenko) [inline submissions/frostwolm/frogger/js/app.js]: 101 is a mysterious number. Please, demystify here and across the code using constants and calculations just like you did for other cases.

File: submissions/frostwolm/frogger/js/app.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+Enemy.prototype.setRndVelocity = function() {
+    var randVel = Enemy.MIN_VELOCITY - 0.5 + Math.random() * (Enemy.MAX_VELOCITY - Enemy.MIN_VELOCITY + 1)
+    randVel = Math.round(randVel);
+    this.velocity = randVel;
+    return this.velocity;
+};
+
+Enemy.prototype.checkCollisions = function(player) {
+    let collisionX = Math.abs((this.x + Enemy.WIDTH) - (player.x + Player.WIDTH)) < Enemy.WIDTH/2;
```
Comment (OleksiyRudenko) [inline submissions/frostwolm/frogger/js/app.js]: Classes should not refer to any global variables, like global variable `player`, which is an instance of `Player` class (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
Hint: pass `Player` instance as an argument to every enemy

Comment (frostwolm) [reply submissions/frostwolm/frogger/js/app.js]: Should I pass an instance of the player as the argument to the Enemy constructor (like I did now) or assign it as a property after creating each enemy object?

Comment (OleksiyRudenko) [reply submissions/frostwolm/frogger/js/app.js]: @frostwolm The implemented approach is an appropriate one 👍

File: submissions/frostwolm/frogger/js/app.js [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  if (this.y > CANVAS_WIDTH - Player.WIDTH + 20) {
+    this.y = CANVAS_WIDTH - Player.WIDTH + 20;
+  }
+};
+
+
+Player.prototype.reset = function(){
+  this.x = 2 * 101;
+  this.y = 5 * 83 - 20;
+  this.direction = null;
```
Comment (OleksiyRudenko) [inline submissions/frostwolm/frogger/js/app.js]: You have constants for some of these numbers. Please, reuse. Here and x-check yourself across entire codebase.
Imagine your peer or collaborator have a task to change imageset (different dimensions, etc). How many places they have to change e.g. cell dimensions at?

File: submissions/frostwolm/frogger/js/app.js [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    this.collisionSubject = collisionSubject;
+    this.velocity = options.velocity || this.setRndVelocity();
+    this.direction = options.direction || DIRECTIONS.RIGHT;
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.HEIGHT = 171;
+Enemy.WIDTH = 101;
```
Comment (OleksiyRudenko) [inline submissions/frostwolm/frogger/js/app.js]: Isn't it strictly bound to the cell height?

**Pull request merged**